app/Mage.php

```diff
@@ -12,45 +12,49 @@
 
 define('DS', DIRECTORY_SEPARATOR);
 define('PS', PATH_SEPARATOR);
-if (file_exists(__DIR__ . '/vendor/mahocommerce/maho')) {
-    define('BP', dirname(getcwd()));
-} elseif (str_contains(__DIR__, '/vendor/mahocommerce/maho')) {
-    define('BP', str_replace('/vendor/mahocommerce/maho/app', '', __DIR__));
+define('BP', str_replace('/vendor/mahocommerce/maho', '', dirname(__DIR__)));
+
+if (file_exists(BP . '/vendor/mahocommerce/maho')) {
+    define('MAHO_IS_STARTER_KIT', true);
+    define('MAHO_ROOT_SOURCE_DIR', BP . '/vendor/mahocommerce/maho');
 } else {
-    define('BP', dirname(__DIR__));
+    define('MAHO_IS_STARTER_KIT', false);
+    define('MAHO_ROOT_SOURCE_DIR', BP);
 }
 
-Mage::register('original_include_path', get_include_path());
-
 if (!empty($_SERVER['MAGE_IS_DEVELOPER_MODE']) || !empty($_ENV['MAGE_IS_DEVELOPER_MODE'])) {
     Mage::setIsDeveloperMode(true);
     ini_set('display_errors', '1');
     ini_set('error_prepend_string', '<pre>');
     ini_set('error_append_string', '</pre>');
+
+    // Fix for overriding zf1-future during development
+    ini_set('opcache.revalidate_path', 1);
 }
 
+require_once MAHO_ROOT_SOURCE_DIR . '/app/code/core/Mage/Core/functions.php';
```

Comment:
the thing is that having MAHO_ROOT_DIR (which already exists) and MAHO_ROOT_SOURCE_DIR is confusing to me.

Reply:
They do point to different locations though. 

`MAHO_ROOT_DIR` is the directory above the public dir.

`MAHO_ROOT_SOURCE_DIR` is the directory of Maho's source.

They're only the same if you git clone. Renaming the constant as you suggest would help I think.